sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -554,18 +554,30 @@ case class JsonToStructs(
   @transient
   lazy val converter = nullableSchema match {
     case _: StructType =>
-      (rows: Seq[InternalRow]) => if (rows.length == 1) rows.head else null
+      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next() else null
```

Comment:
so we may still return null in some cases, can you list them?

Reply:
we don't have to do this PR, but it would be great to document when this expression will return null, in the class doc of `JsonToStructs`

Reply:
... and with some tests to verify it.

Reply:
> we don't have to do it in this PR, but it would be great to document when this expression will return null ...

We already state in the docs for `from_json()`:  `Returns null, in the case of an unparseable string.`

Reply:
For now the case is more concrete, we return null if `Jackson` parser doesn't find any token in the input. Not sure, this detailed info about underlying problem can help users much more.